test/Test/Pos/Util/ModifierSpec.hs

```diff
@@ -0,0 +1,159 @@
+            prop
+                "getting the keys from a map modifier does not eliminate any from the\
+                \ produced list, unless their value was 'Nothing' or they are in the\
+                \ argument list and are a key in the map modifier simultanously"
+                (allKeysAreInMap @Text @Text)
+        describe "values" $ do
+            prop
+                "getting the values from a map modifier does not eliminate any from the\
+                \ produced list, unless their value is 'Nothing' or if it is the value\
+                \ of a key in the input list that does not belong to the the map modifier"
+                (allValsAreInMap @Text @Text)
+        describe "toList" $ do
+            prop
+                "getting the contents of the map modifier as an association list will\
+                \ keep all of the keys whose values are not 'Nothing', and will include\
+                \ the pairs in the argument list whose key is not present in the modifier"
+                (allPairsAreInAssocList @Text @Text)
+        describe "insertionsMap" $ do
+            prop
+                "the created hashmap removes all the 'Nothing' keys from the\
+                \ 'MapModifier' and keeps all the 'Just' keys"
+                (mapModifierToHashMap @Text @Text)
+        describe "insertions" $ do
+            prop
+                "the created list removes all the 'Nothing' keys from the used\
+                \ 'MapModifier' and keeps all the 'Just' keys in the association list"
+                (mapModifierToList @Text @Text)
+        describe "deletions" $ do
+            prop
+                "the created list of keys removes all those with 'Just' values and keeps\
+                \ those with 'Nothing' values"
+                (mapModifierToDeletionList @Text @Text)
+        describe "mapMaybe" $ do
+            prop
+                "the created list of keys removes all those with 'Nothing' values and\
+                \ keeps the pairs from the input list whose keys are not in the modifier"
+                (allPairsAreInMaybeList @Text @Text @Text Just)
+
+
+mapModifierFormsMonoid
+    :: (Show k, Show v, Eq k, Eq v, Hashable k)
+    => Core.MapModifier k v
+    -> Core.MapModifier k v
+    -> Core.MapModifier k v
+    -> Property
+mapModifierFormsMonoid = formsMonoid
+
+insertThenLookup
+    :: (Show k, Show v, Eq k, Eq v, Hashable k)
+    => k
+    -> v
+    -> Core.MapModifier k v
+    -> Bool
+insertThenLookup k v mapMod =
+    let newMap = Core.insert k v mapMod
+    in maybe False (== v) $ Core.lookup (const Nothing) k newMap
+
+insertDeleteThenLookup
+    :: (Show k, Show v, Eq k, Eq v, Hashable k)
+    => k
+    -> v
+    -> Core.MapModifier k v
+    -> Bool
+insertDeleteThenLookup k v mapMod =
+    let newMap@(Core.MapModifier mm) = Core.insert k v mapMod
+        (Core.MapModifier dm) = Core.delete k newMap
+    in (HM.lookup k mm == Just (Just v)) && (HM.lookup k dm == Just Nothing)
```

Comment:
That's not quite what I meant. I could've easiyl put the instance in this module, but I didn't because that's not what we've done so far - `Arbitrary` instance go in separate modules.